### ISSUE TITLE
Implement Option to Change Font

### ISSUE THREAD

**Issue by mariobehling** [fossasia/badgemagic-app#239]
Currently there is only one standard font implemented. Please implement an option to change the font.

**Comment by adityastic** [comment]
@mariobehling this is not possible as of now, we need to have some extra pixels in the badge to morph our letters

**Comment by adityastic** [comment]
@mariobehling @iamareebjamal

**Comment by mariobehling** [comment]
@adityastic This should be possible. It does not depend on the font.

**Comment by adityastic** [comment]
@mariobehling we are talking about the font we send to the device or the font in our application?

**Comment by iamareebjamal** [comment]
We send bitmap to pixel. Where do we send the font?

**Comment by adityastic** [comment]
@iamareebjamal yes, we created hexadecimals of every character, then we convert the input string into characters and fetch the respective hexadecimal for the character. We are not sending the font

**Comment by adityastic** [comment]
By sending the font, I meant the saved hexadecimal codes for every character

**Comment by iamareebjamal** [comment]
Nobody is talking about sending the font. We're talking about converting the font chars to bitmaps

**Comment by fcartegnie** [comment]
That's underestimating work. There's no "magic" way to create bitmap fonts.
Either you rasterize the vector/truetype font (on 1bpp that's really bad) which will look usually bad compared to a hand written bitmap.
Or you use a Unicode Bitmap font, but I doubt you'll find a matching height and that will grow the app.

**Comment by lanrat** [comment]
What about an option to use [BDF](https://en.wikipedia.org/wiki/Glyph_Bitmap_Distribution_Format)  or [PCF](https://en.wikipedia.org/wiki/Portable_Compiled_Format)  bitmap fonts?

I've used them before when driving LEDs.

**Comment by Garv0504** [comment]
hey @mariobehling is this issue is currently open? If open then I would like to work on this.

**Comment by mariobehling** [comment]
@Garv0504 There is a PR open here, but it is not merged yet.

https://github.com/fossasia/badgemagic-app/pull/1194
